rust/pattern/constraint/type_/regex.rs

```diff
@@ -20,14 +20,20 @@
  *
  */
 
-use crate::common::{string::escape_regex, token};
+use crate::common::{string::escape_regex, token, validatable::Validatable, Result};
 use std::fmt;
 
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct RegexConstraint {
     regex: String,
 }
 
+impl Validatable for RegexConstraint {
+    fn validate(&self) -> Result<()> {
+        Ok(()) // TODO validate regex using external crate
+    }
+}
```

Comment:
Not validating in this PR, as this check requires adding the regex or regex-syntax crate to dependencies. This will be done in a separate pull.